src/base/ApplicationLambdaSnsTopicSubscription.ts

```diff
@@ -0,0 +1,139 @@
+import { Resource, TerraformResource } from 'cdktf';
+import { Construct } from 'constructs';
+import { sqs, sns, iam, lambdafunction } from '@cdktf/provider-aws';
+import { SnsTopicSubscriptionConfig } from '@cdktf/provider-aws/lib/sns';
+
+/** The config props type of [[`ApplicationLambdaSnsTopicSubscription]] */
+export interface ApplicationLambdaSnsTopicSubscriptionProps {
+  /** The prefix used to help identify related resources */
+  name: string;
+  /** The SNS topic to subscribe the Lambda to */
+  snsTopicArn: string;
+  /** The Lambda that should be invoked by incoming messages to the SNS topic */
+  lambda: lambdafunction.DataAwsLambdaFunction | lambdafunction.LambdaFunction;
+  /** Tags to apply to the resource(s), where applicable (in this case only the DLQ for the SNS) */
+  tags?: { [key: string]: string };
+  /** Optional list of resource dependencies */
+  dependsOn?: TerraformResource[];
+}
+
+/**
+ * Creates an SNS to Lambda subscription with a DLQ for any messages
+ * that failed to send to the Lambda (e.g. due to permissions error).
+ * Automatically adds policies/permissions for the SNS topic to send
+ * messages to the DLQ and invoke the Lambda function.
+ *
+ * Artifacts:
+ *  * {@link https://www.terraform.io/docs/providers/aws/r/sns_topic_subscription aws_sns_topic_subscription} Resource to subscribe Lambda to SNS topic
+ *  * {@link https://www.terraform.io/docs/providers/aws/r/sqs_queue aws_sqs_queue} Resource (DLQ for the SNS topic)
+ *  * {@link https://www.terraform.io/docs/providers/aws/r/sqs_queue_policy aws_sqs_queue_policy} Resource policy for SNS to send messages to DLQ
+ *  * {@link https://www.terraform.io/docs/providers/aws/r/lambda_permission aws_lambda_permission} Resource permission for SNS to invoke Lambda
+ */
+export class ApplicationLambdaSnsTopicSubscription extends Resource {
+  public readonly snsTopicSubscription: sns.SnsTopicSubscription;
+
+  constructor(
+    scope: Construct,
+    private name: string,
+    private config: ApplicationLambdaSnsTopicSubscriptionProps
+  ) {
+    super(scope, name);
+
+    const snsTopicDlq = this.createSqsSubscriptionDlq();
```

Comment:
can we expose this DLQ to the consuming application? would be useful for the consuming application to pull queue details.

Reply:
Do you envision the application reusing the queue for something else?